scanpy/get.py

```diff
@@ -167,12 +170,20 @@ def obs_df(
         )
 
     # Make df
-    df = pd.DataFrame(index=adata.obs_names)
-    for k, l in zip(keys, lookup_keys):
-        if not use_raw or k in adata.obs.columns:
-            df[k] = adata.obs_vector(l, layer=layer)
-        else:
-            df[k] = adata.raw.obs_vector(l)
+    df = pd.DataFrame(index=adata.obs.index)
+
+    # add var values
+    if len(var_names) > 0:
+        X = _get_obs_rep(adata, layer=layer, use_raw=use_raw)
+        matrix = X[:, adata.var_names.get_indexer(var_names)]
```

Comment:
Do you need to convert `matrix` to a dense array here?

Reply:
needs to be converted to dense if sparse. Just added the code for this.